### PR TITLE
Update binutils to 2.36.1

### PR DESCRIPTION
**Pull request by anphel31**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Update binutils to version 2.36.1 to resolve issue embedding ELF file notes using linker scripts

###### Change Log  

- Change binutils to version 2.36.1. Remove patch files specific to binutils 2.32
- Change kernel's CONFIG_LD_VERSION
- Change qemu-kvm, kexec-tools and tboot to compile with binutils 2.36.1

###### Does this affect the toolchain?  

**YES**
NO

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/...

###### Test Methodology

- Local package build. Pipeline buddybuild

### CODE REVIEW

File: SPECS/tboot/tboot.spec [hunk 1]
```diff
@@ -1,29 +1,27 @@
...
-BuildRequires: trousers-devel
-Requires:      libtspi
+Group:          System Environment/Security
+URL:            https://sourceforge.net/projects/tboot/
+Source0:        http://downloads.sourceforge.net/%{name}/%{name}-%{version}.tar.gz
+BuildRequires:  trousers-devel
+Requires:       libtspi
+ExclusiveArch:  x86_64
 
-Patch0:    no_error_packed_member.patch
```
Comment (PawelWMS) [inline SPECS/tboot/tboot.spec:15]: Did we remove the file as well? I don't see it changed in this PR.

**Review by PawelWMS — APPROVED** [review]

**Review by christopherco — APPROVED** [review]
Kernel package changes LGTM

File: SPECS/kernel/kernel.spec [hunk 2]
```diff
@@ -302,9 +302,9 @@ make INSTALL_MOD_PATH=%{buildroot} modules_install
 # Verify for build-id match
 # We observe different IDs sometimes
 # TODO: debug it
-ID1=`readelf -n vmlinux | grep "Build ID"`
+ID1=`eu-readelf -n vmlinux | grep "Build ID"`
```
Comment (PawelWMS) [inline SPECS/kernel/kernel.spec:305]: Not sure why we're doing this change, but shouldn't we do the same thing inside `kernel-hyperv.spec`? Could you mention this change inside the changelog as well?

File: SPECS/kernel-hyperv/kernel-hyperv.spec [hunk 3]
```diff
@@ -274,6 +274,9 @@ ln -sf linux-%{uname_r}.cfg /boot/mariner.cfg
 %{_libdir}/perf/include/bpf/*
 
 %changelog
+* Mon Mar 29 2021 Andrew Phelps <anphel@microsoft.com> - 5.10.21.1-3
```
Comment (PawelWMS) [inline SPECS/kernel-hyperv/kernel-hyperv.spec:277]: Shouldn't we change the `ID1=`readelf -n vmlinux | grep "Build ID"` lines in this spec the same way it was done for the "regular" kernel?

**Comment by anphel31** [comment]
This PR is replaced by #932

**Pull request closed without merging**